cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -686,6 +691,8 @@ void deleteApp(Id.Application appId, ApplicationSpecification spec) throws Excep
     }
     deleteProgramLocations(appId);
 
+    deleteAppBusinessMetadata(appId);
```

Comment:
this can get called during cdap upgrade. Will everything work in the upgrade tool?